src/BlockTemplatesController.php

```diff
@@ -98,25 +206,63 @@ public function get_block_templates() {
 		$template_files = BlockTemplateUtils::gutenberg_get_template_paths( $this->templates_directory );
 		$templates      = array();
 
+		// Check if the template has been saved in the database first.
+		$check_query_args    = array(
+			'post_type'      => 'wp_template',
+			'posts_per_page' => -1,
+			'no_found_rows'  => true,
+			'tax_query'      => array( // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+				array(
+					'taxonomy' => 'wp_theme',
+					'field'    => 'name',
+					'terms'    => 'woocommerce',
+				),
+			),
+		);
+		$check_query         = new \WP_Query( $check_query_args );
+		$saved_woo_templates = $check_query->posts;
+
+		// From a WordPress post, build the template object and add it to our list.
+		foreach ( $saved_woo_templates as $saved_woo_template ) {
+			$templates[] = BlockTemplateUtils::gutenberg_build_template_result_from_post( $saved_woo_template );
+		}
+
 		foreach ( $template_files as $template_file ) {
```

Comment:
`get_block_templates_from_theme` won't be necessary because at this point the theme templates are already loaded.

I did split it into `get_block_templates_from_db` and `get_block_templates_from_woocommerce` though, I also had to introduce a new function: `remove_theme_templates_with_custom_alternative` because in the following scenario, duplicate pages would be available in the editor:

1. Theme has no template, but WooCommerce does
2. User edits the template and saves it to DB.
3. Theme updates and includes template.
4. Without this change, the theme would include the template and woo would also include it, from the db causing a duplicate,